ddl/job_table.go

```diff
@@ -309,6 +316,160 @@ func (d *ddl) markJobProcessing(sess *session, job *model.Job) error {
 	return errors.Trace(err)
 }
 
+func (d *ddl) startDispatchBackfillJobsLoop() {
+	isDistReorg := variable.DDLEnableDistributeReorg.Load()
+	if !isDistReorg {
+		return
+	}
+	d.backfillCtx.jobCtxMap = make(map[int64]*JobContext)
+	d.backfillCtx.backfillCtxMap = make(map[int64]struct{})
+
+	var notifyBackfillJobByEtcdCh clientv3.WatchChan
+	if d.etcdCli != nil {
+		notifyBackfillJobByEtcdCh = d.etcdCli.Watch(d.ctx, addingBackfillJob)
+	}
+	ticker := time.NewTicker(500 * time.Millisecond)
+	defer ticker.Stop()
+	for {
+		if isChanClosed(d.ctx.Done()) {
+			return
+		}
+		select {
+		case <-d.backfillJobCh:
+		case <-ticker.C:
+		case _, ok := <-notifyBackfillJobByEtcdCh:
+			if !ok {
+				logutil.BgLogger().Warn("[ddl] start backfill worker watch channel closed", zap.String("watch key", addingBackfillJob))
+				notifyBackfillJobByEtcdCh = d.etcdCli.Watch(d.ctx, addingBackfillJob)
+				time.Sleep(time.Second)
+				continue
+			}
+		case <-d.ctx.Done():
+			return
+		}
+		d.loadBackfillJobAndRun()
+	}
+}
+
+func (d *ddl) getTableByTxn(store kv.Storage, schemaID, tableID int64) (*model.DBInfo, table.Table, error) {
+	var tbl table.Table
+	var dbInfo *model.DBInfo
+	err := kv.RunInNewTxn(d.ctx, store, false, func(ctx context.Context, txn kv.Transaction) error {
+		t := meta.NewMeta(txn)
+		var err1 error
```

Comment:
I think we could reuse `err`

Reply:
It's just easy to read. It's a little bit more clear.